crates/biome_graphql_parser/tests/graphql_test_suite/err/directive.graphql.snap

```diff
@@ -0,0 +1,821 @@
+
+{
+  hero @(reason: "Deprecated"
+}
+
+{
+  hero @(: "Deprecated"
+}
+
+{
+  hero @(:
+}
+
+{
+  hero @(:)
+}
+
+```
+
+## AST
+
+```
+GraphqlRoot {
+    bom_token: missing (optional),
+    definitions: GraphqlDefinitionList [
+        GraphqlSelectionSet {
+            l_curly_token: L_CURLY@0..1 "{" [] [],
+            selections: GraphqlSelectionList [
+                GraphqlField {
+                    alias: missing (optional),
+                    name: GraphqlName {
+                        value_token: GRAPHQL_NAME@1..9 "hero" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
+                    },
+                    arguments: missing (optional),
+                    directives: GraphqlDirectiveList [
+                        GraphqlDirective {
+                            at_token: AT@9..10 "@" [] [],
+                            name: missing (required),
+                            arguments: missing (optional),
+                        },
+                    ],
+                    selection_set: missing (optional),
+                },
+            ],
+            r_curly_token: R_CURLY@10..12 "}" [Newline("\n")] [],
+        },
+        GraphqlSelectionSet {
+            l_curly_token: L_CURLY@12..15 "{" [Newline("\n"), Newline("\n")] [],
+            selections: GraphqlSelectionList [
+                GraphqlField {
+                    alias: missing (optional),
+                    name: GraphqlName {
+                        value_token: GRAPHQL_NAME@15..23 "hero" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
+                    },
+                    arguments: missing (optional),
+                    directives: GraphqlDirectiveList [
+                        GraphqlDirective {
+                            at_token: AT@23..25 "@" [] [Whitespace(" ")],
+                            name: missing (required),
+                            arguments: missing (optional),
+                        },
+                        GraphqlDirective {
+                            at_token: AT@25..26 "@" [] [],
+                            name: missing (required),
+                            arguments: missing (optional),
+                        },
+                    ],
+                    selection_set: missing (optional),
+                },
+            ],
+            r_curly_token: R_CURLY@26..28 "}" [Newline("\n")] [],
+        },
+        GraphqlSelectionSet {
+            l_curly_token: L_CURLY@28..31 "{" [Newline("\n"), Newline("\n")] [],
+            selections: GraphqlSelectionList [
+                GraphqlBogusSelection {
+                    items: [
+                        GraphqlName {
+                            value_token: GRAPHQL_NAME@31..39 "hero" [Newline("\n"), Whitespace("  ")] [Whitespace(" ")],
+                        },
+                        GraphqlBogus {
+                            items: [
+                                GraphqlBogus {
+                                    items: [
```

Comment:
The reason there is a nested bogus nodes here was because in this case:
```graphql
{
  hero @deprecated(: "Deprecated")
}
```
The parser tried to parse `(: "Deprecated")` as an `GraphqlArgumentList`, but since this argument list has an invalid argument `: "Deprecated"` the entire list was turn into a bogus node, which caused `@deprecated(: "Deprecated")` to turn from a `GraphqlDirective` to a bogus node, which turned `hero @deprecated(: "Deprecated")` from a `GraphqlField` to a bogus node, caused a series of nested bogus nodes. Do you have any suggestions on how to improve this?